lib/ash_authentication.ex

```diff
@@ -1,101 +1,23 @@
-          The subject name is used in generating token claims and in generating
-          authentication routes.
-
-          This needs to be unique system-wide and if not set will be inferred
-          from the resource name (ie `MyApp.Accounts.User` will have a subject
-          name of `user`).
-          """
-        ],
-        api: [
-          type: {:behaviour, Ash.Api},
-          doc: """
-          The name of the Ash API to use to access this resource when
-          registering/authenticating.
-          """,
-          required: true
-        ],
-        get_by_subject_action_name: [
-          type: :atom,
-          doc: """
-          The name of the read action used to retrieve records.
-
-          Used internally by `AshAuthentication.subject_to_resource/2`.  If the
-          action doesn't exist, one will be generated for you.
-          """,
-          default: :get_by_subject
-        ]
-      ]
-    },
-    %Spark.Dsl.Section{
-      name: :tokens,
-      describe: "Configure JWT settings for this resource",
-      schema: [
-        enabled?: [
-          type: :boolean,
-          doc: """
-          Should JWTs be generated by this resource?
-          """,
-          default: false
-        ],
-        signing_algorithm: [
-          type: :string,
-          doc: """
-          The algorithm to use for token signing.
-
-          Available signing algorithms are;
-          #{to_sentence(Joken.Signer.algorithms(), final: "and")}.
-          """,
-          default: hd(Joken.Signer.algorithms())
-        ],
-        token_lifetime: [
-          type: :pos_integer,
-          doc: """
-          How long a token should be valid, in hours.
-          """
-        ],
-        revocation_resource: [
-          type: {:behaviour, Ash.Resource},
-          doc: """
-          The resource used to store token revocation information.
-
-          If token generation is enabled for this resource, we need a place to
-          store revocation information. This option is the name of an Ash
-          Resource which has the `AshAuthentication.TokenRevocation` extension
-          present.
-          """
-        ]
-      ]
-    }
-  ]
+  require AshAuthentication.Dsl
+  import AshAuthentication.Dsl
 
   @moduledoc """
   AshAuthentication provides a turn-key authentication solution for folks using
   [Ash](https://www.ash-hq.org/).
 
   ## Usage
 
-  This package assumes that you have [Phoenix](https://phoenixframework.org/) and
-  [Ash](https://ash-hq.org/) installed and configured.  See their individual
-  documentation for details.
+  This package assumes that you have [Ash](https://ash-hq.org/) installed and
+  configured.  See the Ash documentation for details.
 
-  Once installed you can easily add support for authentication by configuring one
-  or more extensions onto your Ash resource:
+  Once installed you can easily add support for authentication by configuring
+  one or more extensions onto your Ash resource:
```

Comment:
should this now just say something like "by configuring the AshAuthentication extension on your resource"